ansible_ai_connect/ai/api/views.py

```diff
@@ -1012,3 +1024,90 @@ def write_to_segment(
                 },
             )
         send_segment_event(event, "codegenPlaybook", user)
+
+
+class Chat(APIView):
+    """
+    Send a message to the backend chatbot service and get a reply.
+    """
+
+    permission_classes = [
+        permissions.IsAuthenticated,
+        IsAuthenticatedOrTokenHasScope,
+    ]
+    required_scopes = ["read", "write"]
+
+    CHAT_API_NOT_ENABLED = "Chat API is not enabled."
+    INVALID_CHAT_REQUEST = "An invalid chat request was received."
+    INVALID_CHAT_RESPONSE = "An invalid chat response was received."
+
+    def __init__(self):
+        self.chatbot_enabled = (
+            settings.CHATBOT_URL
+            and settings.CHATBOT_DEFAULT_MODEL
+            and settings.CHATBOT_DEFAULT_PROVIDER
+        )
+        if self.chatbot_enabled:
+            logger.debug("Chatbot is enabled.")
+        else:
+            logger.debug("Chatbot is not enabled.")
+
+    @extend_schema(
+        request=ChatRequestSerializer,
+        responses={
+            200: ChatResponseSerializer,
+            400: OpenApiResponse(description="Bad Request"),
+        },
+        summary="Chat request",
+    )
+    def post(self, request) -> Response:
+        headers = {"Content-Type": "application/json"}
+        request_serializer = ChatRequestSerializer(data=request.data)
+
+        try:
+            if not self.chatbot_enabled:
+                raise ChatbotNotEnabledException()
+
+            if not request_serializer.is_valid():
+                raise ChatbotInvalidRequestException()
+
+            data = {
+                "query": request_serializer.validated_data["query"],
+                "model": settings.CHATBOT_DEFAULT_MODEL,
+                "provider": settings.CHATBOT_DEFAULT_PROVIDER,
+            }
+            if "conversation_id" in request_serializer.validated_data:
+                data["conversation_id"] = request_serializer.validated_data["conversation_id"]
+            response = requests.post(settings.CHATBOT_URL + "/v1/query", headers=headers, json=data)
+
+            if response.status_code == 200:
+                data = json.loads(response.text)
+                response_serializer = ChatResponseSerializer(data=data)
+
+                if not response_serializer.is_valid():
+                    raise ChatbotInvalidResponseException()
+
+                return Response(
+                    data,
+                    status=rest_framework_status.HTTP_200_OK,
+                    headers=headers,
+                )
+            elif response.status_code == 401:
```

Comment:
These set of status codes (`401`, `403`, `413`, ...) are the ones defined in the Open API schema for OpenShift lightspeed-service.